src/reporters/base.ts

```diff
@@ -345,30 +344,26 @@ export abstract class BaseReporter {
    */
   async priceToReport() {
     const getPrice = async (): Promise<BigNumber> => {
-      // If the circuit breaker is already open, complain loudly
-      if (this.circuitBreakerOpen) {
-        throw Error('Circuit breaker is open')
-      }
       // This can throw if there is an issue with trade data
       const price = await this.config.dataAggregator.currentPrice()
-      // lastReportedPrice is a BigNumber and therefore truthy even if zero.
-      // Determine if we should open the circuit breaker.
-      // TODO: consider lastReportedTime in circuit breaker calculation
-
-      this._circuitBreakerOpen =
-        (this.calculateCircuitBreakerPriceChangeThreshold() !== undefined &&
-          this.lastReportedPrice &&
-          isOutsideTolerance(
-            this.lastReportedPrice,
-            price,
-            this.calculateCircuitBreakerPriceChangeThreshold()
-          )) ||
-        false // false so that it can't be undefined
-
-      if (this._circuitBreakerOpen) {
-        throw Error(
-          `Opening circuit breaker, price to report is too different from the last reported price. Price: ${price} Last reported price: ${this.lastReportedPrice} Price change threshold: ${this.config.circuitBreakerPriceChangeThresholdMin}`
-        )
+
+      // Circuit breaker logic only applies if the oracle client has previously
+      // reported.
+      const haveReported =
+        this.lastReportedTimeMs !== undefined && this.lastReportedPrice !== undefined
+      if (haveReported) {
+        // Determine if we should open the circuit breaker.
+        const circuitBreakerThreshold = this.calculateCircuitBreakerPriceChangeThreshold()
+        const timeSinceLastReport = Date.now() - this.lastReportedTimeMs!
+        const circuitBreakerOpen =
+          isOutsideTolerance(this.lastReportedPrice!, price, circuitBreakerThreshold) &&
+          timeSinceLastReport < this.config.circuitBreakerDurationTimeMs
```

Comment:
Just an idea here: Can we add some logic so that if circuitBreakerDurationTimeMs*3 has already passed and the circuit breaker triggers again, the oracles now wait longer to report again?

Reply:
I'm not sure I follow. Do you mean to wait longer each time it triggers? What it a longe time has passed since the last time it has triggered?

Reply:
yeah, exactly

Reply:
As discussed, if such an increasing circuit breaker duration is to be implemented, it'll be done on follow-up PRs.